elastic/apm-server
# apmbench: Warm-up with specified agent number

## Motivation/summary

Changes the benchmark warm-up to use the same amount of specified agents
and fixes the broken glob expression, causing warm-up to be skipped.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Build `apmbench`: `cd systemtest/cmd/apmbench && go build .`
2. Run it with an `-agents` > 1: `./apmbench -agents=5 -run BenchmarkAgent`
3. Navigate to `localhost:8200/debug/expvar`, check that the total number of `acked` events is > `25000` (5000 warm up events by 5 agents).

## Related issues

Part of #7540
